src/code42cli/cmds/devices.py

```diff
@@ -0,0 +1,399 @@
+from collections import OrderedDict
+from datetime import date
+from datetime import datetime
+
+import click
+from pandas import DataFrame
+from pandas import to_datetime
+from pandas import to_timedelta
+from py42 import exceptions
+
+from code42cli.bulk import run_bulk_process
+from code42cli.click_ext.groups import OrderedGroup
+from code42cli.errors import Code42CLIError
+from code42cli.file_readers import read_csv_arg
+from code42cli.options import format_option
+from code42cli.options import sdk_options
+from code42cli.output_formats import OutputFormatter
+
+
+@click.group(cls=OrderedGroup)
+@sdk_options(hidden=True)
+def devices(state):
+    """For managing computers within Code42."""
+    pass
+
+
+device_id_option = click.option(
+    "--device-id",
+    required=True,
+    type=str,
+    help="The device ID for the device on which you wish to act.",
+)
+
+change_device_name_option = click.option(
+    "--change-device-name",
+    required=False,
+    is_flag=True,
+    default=False,
+    help='Include to prepend "deactivated_" and today\'s date to the name of any deactivated devices.',
+)
+
+purge_date_option = click.option(
+    "--purge-date",
+    required=False,
+    type=str,
+    default=None,
+    help="The date on which the archive should be purged from cold storage in yyyy-MM-dd format. If not provided, the date will be set according to the appropriate org settings.",
+)
+
+
+@devices.command()
+@device_id_option
+@change_device_name_option
+@purge_date_option
+@sdk_options()
+def deactivate(state, device_id, change_device_name, purge_date):
```

Comment:
`device_id` should probably be an `click.Argument` rather than a `click.option` since it is always required and there are no other required args, that is the pattern we have been doing anyway.

Reply:
I've updated this to an argument.